core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java

```diff
@@ -133,19 +134,18 @@ public void init() {
         timerExecutor.scheduleAtFixedRate(new Runnable() {
             @Override
             public void run() {
-                List<MessageFuture> timeoutMessageFutures = new ArrayList<MessageFuture>(futures.size());
-                for (MessageFuture future : futures.values()) {
-                    if (future.isTimeout()) {
-                        timeoutMessageFutures.add(future);
-                    }
-                }
-                for (MessageFuture messageFuture : timeoutMessageFutures) {
-                    futures.remove(messageFuture.getRequestMessage().getId());
-                    messageFuture.setResultMessage(null);
-                    if (LOGGER.isDebugEnabled()) {
-                        LOGGER.debug("timeout clear future : " + messageFuture.getRequestMessage().getBody());
+                Iterator<Map.Entry<Integer, MessageFuture>> futuresIterator = futures.entrySet().iterator();
+                for (; futuresIterator.hasNext(); ) {
```

Comment:
Use while since there is only one condition

Reply:
misunderstood what I meant，
```
for(Map.Entry<Integer,MessageFuture> entry: futures.entrySet()){xxx}